packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -587,3 +587,403 @@ class MultiTapGestureRecognizer extends GestureRecognizer {
+  /// [onSerialTapCancel] nor [onSerialTapUp] have yet fired.
+  bool get isTrackingPointer => _pendingTap != null;
+
+  @override
+  bool isPointerAllowed(PointerDownEvent event) {
+    if (_completedTaps.isEmpty) {
+      if (onSerialTapDown == null && onSerialTapCancel == null && onSerialTapUp == null) {
+        return false;
+      }
+    }
+    return super.isPointerAllowed(event);
+  }
+
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    if (_completedTaps.isNotEmpty && !_representsSameSeries(_completedTaps.last, event)) {
+      _reset();
+    } else if (_pendingTap != null) {
+      _reset();
+    }
+    _trackTap(event);
+  }
+
+  bool _representsSameSeries(_TapTracker tap, PointerDownEvent event) {
+    return tap.hasElapsedMinTime() // touch screens often detect touches intermittently
+        && tap.hasSameButton(event)
+        && tap.isWithinGlobalTolerance(event, kDoubleTapSlop);
+  }
+
+  void _trackTap(PointerDownEvent event) {
+    _stopDoubleTapTimer();
+    if (onSerialTapDown != null) {
+      final SerialTapDownDetails details = SerialTapDownDetails(
+        globalPosition: event.position,
+        localPosition: event.localPosition,
+        kind: getKindForPointer(event.pointer),
+        buttons: event.buttons,
+        count: _completedTaps.length + 1,
+      );
+      invokeCallback<void>('onSerialTapDown', () => onSerialTapDown!(details));
+    }
+    final _TapTracker tracker = _TapTracker(
+      event: event,
+      entry: GestureBinding.instance!.gestureArena.add(event.pointer, this),
+      doubleTapMinTime: kDoubleTapMinTime,
+    );
+    assert(_pendingTap == null);
+    _pendingTap = tracker;
+    tracker.startTrackingPointer(_handleEvent, event.transform);
+  }
+
+  void _handleEvent(PointerEvent event) {
+    assert(_pendingTap != null);
+    assert(_pendingTap!.pointer == event.pointer);
+    final _TapTracker tracker = _pendingTap!;
+    if (event is PointerUpEvent) {
+      _registerTap(event, tracker);
+    } else if (event is PointerMoveEvent) {
+      if (!tracker.isWithinGlobalTolerance(event, kDoubleTapTouchSlop)) {
+        _reset();
+      }
+    } else if (event is PointerCancelEvent) {
+      _reset();
+    }
+  }
+
+  @override
+  void acceptGesture(int pointer) { }
+
+  @override
+  void rejectGesture(int pointer) {
+    _reset();
+  }
+
+  void _rejectPendingTap() {
+    assert(_pendingTap != null);
+    final _TapTracker tracker = _pendingTap!;
+    _pendingTap = null;
+    // Order is important here; the `resolve` call can yield a re-entrant
+    // `reset()`, so we need to check cancel here while we can trust the
+    // length of our _completedTaps list.
+    _checkCancel(_completedTaps.length + 1);
+    tracker.entry.resolve(GestureDisposition.rejected);
+    _stopTrackingPointer(tracker);
+  }
+
+  @override
+  void dispose() {
+    if (_pendingTap != null) {
+      _rejectPendingTap();
+    }
```

Comment:
`_rest` called below already includes this.

Reply:
Done